web/frontend/angular/src/app/main-pages/register/register.component.ts

```diff
@@ -36,7 +38,20 @@ export class RegisterComponent implements OnInit {
         if (this.authForm.invalid) {
             return
         }
-        this.authService.register(this.authForm.value)
-        this.router.navigateByUrl("homepage")
+        this.authService.register(this.authForm.value).subscribe(
+            (res: any) => {
+                console.log(res)
+                this.router.navigateByUrl("homepage")
+            },
+            (error) => {
```

Comment:
You need to clear all the error stuff if a request succeeds. Currently, if the first request fails but the second succeeds, everything still appears red.

There should be some feedback to the user about successful registration. Like a message somewhere that says registration was successful.

Reply:
This clears some errors, but not all of them.

![bild](https://user-images.githubusercontent.com/1515135/163878986-6f30aa83-e072-497f-b1c8-a08cf69036a6.png)

I initially had two errors: the password being short, and the passwords not matching. The short password error disappeared, but as you can see, the passwords not matching error remains.